### ISSUE TITLE
False positive `-Winvalid-noreturn` if `throw` is followed by `try`/`catch`

### ISSUE THREAD

**Issue by aaronpuchert** [llvm/llvm-project#174822]
Compile with `-fsyntax-only -Winvalid-noreturn`:

```c++
[[noreturn]] void test() {
    throw 1;
    try {} catch(...) {}
}
```

This produces:
```
<source>:4:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
    4 | }
      | ^
```

I think that's because we don't add exception handling edges for calls to the CFG, so we mark all catch blocks as reachable. But this is just speculation.
